internal/provider/network_connection_models.go

```diff
@@ -0,0 +1,289 @@
+					"peer_network_url": resourceschema.StringAttribute{
+						Required:            true,
+						MarkdownDescription: "The URL of the peer network",
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.RequiresReplace(),
+						},
+					},
+					"managed_network_url": resourceschema.StringAttribute{
+						Computed:            true,
+						MarkdownDescription: "The URL of the managed GCP network",
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
+						},
+					},
+				},
+				Validators: []validator.Object{
+					objectvalidator.ExactlyOneOf(path.Expressions{
+						path.MatchRoot("aws"),
+						path.MatchRoot("google"),
+					}...),
+				},
+			},
+		},
+	}
+)
+
+func generateNetworkConnectionDatasourceSchema(diagn diag.Diagnostics) dataschema.Schema {
+	res := dataschema.Schema{
+		MarkdownDescription: "Network Connection datasource",
+		Attributes:          convertSchemaAttributes(networkConnectionResourceSchema.Attributes, diagn),
+	}
+
+	id := res.Attributes["id"].(*dataschema.StringAttribute)
+	id.Computed = false
+	id.Required = true
+
+	return res
+}
+
+func (m *NetworkConnectionModel) FromProtobuf(nc *network.NetworkConnection) error {
+	m.NetworkID = types.StringValue(nc.NetworkId)
+	m.ID = types.StringValue(nc.Id)
+	m.Description = types.StringValue(nc.Description)
+
+	switch {
+	case nc.GetAws() != nil:
+		if m.AWS == nil {
+			m.AWS = &awsNetworkConnectionInfo{}
+		}
+		switch {
+		case nc.GetAws().GetPeering() != nil:
+			if m.AWS.Peering == nil {
+				m.AWS.Peering = &awsNetworkConnectionPeeringInfo{}
+			}
+			params := nc.GetAws().GetPeering()
+			peering := m.AWS.Peering
+
+			peering.PeeringConnectionID = types.StringValue(params.PeeringConnectionId)
+			peering.ManagedIPv4CIDRBlock = types.StringValue(params.ManagedIpv4CidrBlock)
+			peering.ManagedIPv6CIDRBlock = types.StringValue(params.ManagedIpv6CidrBlock)
+			peering.VPCID = types.StringValue(params.VpcId)
+			peering.AccountID = types.StringValue(params.AccountId)
+			peering.RegionID = types.StringValue(params.RegionId)
+			peering.IPv4CIDRBlock = types.StringValue(params.Ipv4CidrBlock)
+			peering.IPv6CIDRBlock = types.StringValue(params.Ipv6CidrBlock)
+		default:
+			return fmt.Errorf("unsupported type of AWS connection: %v", nc.GetAws().GetType())
+		}
+	case nc.GetGoogle() != nil:
+		if m.Google == nil {
+			m.Google = &googleNetworkConnectionInfo{}
+		}
+		params := nc.GetGoogle()
+		peering := m.Google
+
+		peering.Name = types.StringValue(params.Name)
+		peering.PeerNetworkURL = types.StringValue(params.PeerNetworkUrl)
+		peering.ManagedNetworkURL = types.StringValue(params.ManagedNetworkUrl)
+	default:
+		return fmt.Errorf("unsupported type of network connection: %v", nc.GetConnectionInfo())
+	}
+
+	return nil
+}
+
+func getNetworkConnection(
+	ctx context.Context,
+	client *dcgennet.NetworkConnectionServiceClient,
+	id string,
+	data *NetworkConnectionModel,
+	diagn diag.Diagnostics,
+) bool {
```

Comment:
It's a little bit strange to return `bool` on `get` request.
Can we return `(model, diag.Diagnostics)` in method like this?